From ed708111f2ad92f83a9c3bce7767afe59134425b Mon Sep 17 00:00:00 2001
From: Jonathan Hsieh <jon@cloudera.com>
Date: Thu, 21 Jul 2011 11:53:39 -0700
Subject: [PATCH 41/65] FLUME-716: Add ability to exclude or include arbitrary tests to maven build

---
 DEVNOTES |   53 +++++++++++++++++++++++++++++++++++++++++++++++++++++
 pom.xml  |   11 +++++++++++
 2 files changed, 64 insertions(+), 0 deletions(-)

diff --git a/DEVNOTES b/DEVNOTES
index 51018d8..0355bb7 100644
--- a/DEVNOTES
+++ b/DEVNOTES
@@ -163,6 +163,59 @@ different directories and thus need to change the setting.
 </settings>
 ----
 
+==== Including or excluding specific sets of tests.
+
+We've added hooks to the maven build that will enable you to exclude
+or include specific tests on a test run.  This is useful for excluding
+flakey tests or making a build that focuses solely upon flakey tests.
+
+To do this we created two variables:
+
+# test.include.pattern
+# test.exclude.pattern
+
+These variables take regular expression patterns of the files to be
+included or excluded.
+
+For the next set of examples, let's say you have flakey test called
+TestFlaky1 and TestFlaky2.
+
+You can execute tests that skip TestFlaky1 and TestFlaky2 by using the
+following command line:
+
+----
+mvn test -Dtest.exclude.pattern=**/TestFlaky*.java
+----
+
+Alternately, you could be more explicit
+
+----
+mvn test -Dtest.exclude.pattern=**/TestFlaky1.java,**/TestFlaky2.java
+----
+
+Conversely, you could execute only the flaky tests by using:
+
+----
+mvn test -Dtest.include.pattern=**/TestFlaky*.java
+----
+
+You can also have a combination of imports and exports.  This runs
+TestFlaky* but skips over TestFlaky2:
+
+----
+mvn test -Dtest.include.pattern=**/TestFlaky*.java -Dtest.exclude.pattern=**/TestFlaky2.java
+----
+
+NOTE: Both test.exclude.pattern and test.include.pattern get
+overridden if the test parameter is used.  Consider:
+
+----
+mvn test -Dtest.exclude.pattern=**/TestFlaky*.java -Dtest=TestFlaky1
+---
+
+In this case, TestFlaky1 will be run despite being in the
+test.exclude.pattern.
+
 === Running the most recent build
 
 To run the most recent build of Flume, first build the distribuion
diff --git a/pom.xml b/pom.xml
index 3f39343..e341607 100644
--- a/pom.xml
+++ b/pom.xml
@@ -22,6 +22,10 @@
 
     <!-- Set default encoding to UTF-8 to remove maven complaints -->
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
+    <!-- defaults for flaky test and focused test exclusions -->
+    <test.exclude.pattern>$</test.exclude.pattern> <!-- junk pattern -->
+    <test.include.pattern>**/Test*.java</test.include.pattern>
   </properties>
 
   <profiles>
@@ -381,6 +385,13 @@
             <forkMode>always</forkMode>
             <forkedProcessTimeoutInSeconds>480</forkedProcessTimeoutInSeconds>
             <redirectTestOutputToFile>true</redirectTestOutputToFile>
+	    <includes>
+	      <include>${test.include.pattern}</include>
+	    </includes>
+	    <excludes>
+	      <exclude>**/*$*</exclude>
+	      <exclude>${test.exclude.pattern}</exclude>
+	    </excludes>
           </configuration>
         </plugin>
 
-- 
1.7.0.4

