From 7095bf0e556b43ced7112b777baa31984efeecbf Mon Sep 17 00:00:00 2001
From: Jonathan Hsieh <jon@cloudera.com>
Date: Fri, 23 Sep 2011 15:08:02 -0700
Subject: [PATCH 53/65] CLOUDERA-BUILD: Include git hash properly in do-release-build

---
 cloudera/do-release-build |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/cloudera/do-release-build b/cloudera/do-release-build
index d18af6f..072ef58 100755
--- a/cloudera/do-release-build
+++ b/cloudera/do-release-build
@@ -73,7 +73,7 @@ fi
 #echo
 
 # Add -Pfull-build to include documentation build
-mvn clean source:jar install javadoc:javadoc -Dmaven.revision=${FLUME_GIT_HASH} -Dthrift.executable=${THRIFT_HOME}/bin/thrift -DskipTests -Pfull-build ${DO_MAVEN_DEPLOY}
+mvn clean source:jar install javadoc:javadoc -Dbuild.revision=${FLUME_GIT_HASH} -Dthrift.executable=${THRIFT_HOME}/bin/thrift -DskipTests -Pfull-build ${DO_MAVEN_DEPLOY}
 
 mkdir build
 cp flume-distribution/target/flume-distribution-${FULL_VERSION}-project.tar.gz build/flume-${FULL_VERSION}.tar.gz
-- 
1.7.0.4

